handlers/handlers.go

```diff
@@ -31,6 +32,143 @@ func Read(cfg *config.Config, zc ZebedeeClient, rend RenderClient, searchC Searc
+	if err != nil && !errs.ErrMapForRenderBeforeAPICalls[err] {
+		log.Error(ctx, "unable to review query", err)
+		setStatusCode(w, req, err)
+		return
+	}
+
+	var (
+		// counter used to keep track of the number of concurrent API calls
+		counter            = 3
+		errorMessage       string
+		makeSearchAPICalls = true
+	)
+
+	// avoid making unecessary search API calls
+	if errs.ErrMapForRenderBeforeAPICalls[err] {
+		makeSearchAPICalls = false
+
+		// reduce counter by the number of concurrent search API calls that would be
+		// run in go routines
+		counter -= 2
+		errorMessage = err.Error()
+	}
+
+	searchQuery := data.GetSearchAPIQuery(validatedQueryParams, censusTopicCache)
+	categoriesCountQuery := getCategoriesCountQuery(searchQuery)
+
+	var (
+		homepageResp zebedeeCli.HomepageContent
+		searchResp   = &searchModels.SearchResponse{}
+
+		categories      []data.Category
+		topicCategories []data.Topic
+		populationTypes []data.PopulationTypes
+		dimensions      []data.Dimensions
+
+		wg sync.WaitGroup
+
+		respErr, countErr error
+	)
+
+	wg.Add(counter)
+
+	go func() {
+		defer wg.Done()
+		var homeErr error
+		homepageResp, homeErr = zc.GetHomepageContent(ctx, accessToken, collectionID, lang, homepagePath)
+		if homeErr != nil {
+			log.Warn(ctx, "unable to get homepage content", log.FormatErrors([]error{err}))
+			return
+		}
+	}()
+	if makeSearchAPICalls {
+		var options searchSDK.Options
+
+		options.Query = searchQuery
+		options.Headers = http.Header{
+			searchSDK.Authorization: {"Bearer " + accessToken},
+			searchSDK.CollectionID:  {collectionID},
+		}
+		go func() {
+			defer wg.Done()
+
+			searchResp, respErr = searchC.GetSearch(ctx, options)
+			if respErr != nil {
+				log.Error(ctx, "getting search response from client failed", respErr)
+				cancel()
+				return
+			}
+		}()
+		go func() {
+			defer wg.Done()
+
+			// TO-DO: Need to make a second request until API can handle aggregration on datatypes (e.g. bulletins, article) to return counts
+			categories, topicCategories, populationTypes, dimensions, countErr = getCategoriesTypesCount(ctx, accessToken, collectionID, categoriesCountQuery, searchC, censusTopicCache)
+			if countErr != nil {
+				log.Error(ctx, "getting categories, types and its counts failed", countErr)
+				setStatusCode(w, req, countErr)
+				cancel()
+				return
+			}
+		}()
+	}
+
+	wg.Wait()
+	if respErr != nil || countErr != nil {
+		setStatusCode(w, req, respErr)
+		return
+	}
+
+	err = validateCurrentPage(ctx, cfg, validatedQueryParams, searchResp.Count)
+	if err != nil {
+		log.Error(ctx, "unable to validate current page", err)
+		setStatusCode(w, req, err)
+		return
+	}
+
+	basePage := rend.NewBasePageModel()
+	m := mapper.CreateDataFinderPage(cfg, req, basePage, validatedQueryParams, categories, topicCategories, populationTypes, dimensions, searchResp, lang, homepageResp, errorMessage, navigationCache)
+	rend.BuildPage(w, m, "search")
+}
```

Comment:
I thought this would render the dataset finder template? I see the `search` templates has been updated but wondering why there isn't a separate dataset finder template.

Reply:
The note on the ticket was to save LOC where possible.  None of the features I'm aware of really change the template enough to justify forking it, but if the consensus is it's easier to understand duplicated, I'm happy to do so.

Reply:
The ticket says there should be a new template (copied from existing), handler and component tests. Re the refactoring bit to save lines of code... maybe some of the page components could be partials? Linden just did some refactoring in the homepage controller. Can find it if you'd like an example.